### PR TITLE
Retry pulling builder image

### PR DESCRIPTION
**Pull request by akalenyu**
**What this PR does / why we need it**:
We flake very frequently on pulling the builder images, let's see if retrying a few times helps:
https://prow.ci.kubevirt.io/view/gs/kubevirt-prow/pr-logs/pull/kubevirt_containerized-data-importer/3371/pull-containerized-data-importer-e2e-hpp-previous/1823003234072006656

Some container runtimes have a `--retry`, other's don't, so opted for writing the retry logic ourselves:
https://docs.podman.io/en/latest/markdown/podman-pull.1.html https://docs.docker.com/reference/cli/docker/image/pull/

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: hack/build/bazel-docker.sh [hunk 1]
```diff
@@ -44,6 +44,13 @@ if [ -z "$(${CDI_CRI} volume list | grep ${BUILDER_VOLUME})" ]; then
     ${CDI_CRI} volume create ${BUILDER_VOLUME}
 fi
 
+# Wait until builder image is pulled
+pullcount=0
+until ${CDI_CRI} pull ${BUILDER_IMAGE}; do
+    ((pullcount++)) && ((pullcount == 5)) && echo "Couldn't pull builder image" && exit 1
+    sleep 1
```
Comment (mhenriks) [inline hack/build/bazel-docker.sh:51]: maybe do an exponential backoff `2 ** pullcount` or something?

Comment (akalenyu) [reply hack/build/bazel-docker.sh]: yep added

**Comment by awels** [comment]
/lgtm
Hopefully this helps, its annoying having to re-run the entire lane because of silly failures to pull the builder.

**Comment by akalenyu** [comment]
> /lgtm Hopefully this helps, its annoying having to re-run the entire lane because of silly failures to pull the builder.

It does, check https://github.com/kubevirt/containerized-data-importer/pull/3387#issuecomment-2288697011

**Pull request merged**